lib/string/sprintf.h

```diff
@@ -53,4 +64,31 @@ xvasprintf(char **restrict s, const char *restrict fmt, va_list ap)
 }
 
 
+inline int
+snprintf_(char *restrict s, int size, const char *restrict fmt, ...)
+{
+	int      len;
+	va_list  ap;
+
+	va_start(ap, fmt);
+	len = vsnprintf_(s, size, fmt, ap);
+	va_end(ap);
+
+	return len;
+}
+
+
+inline int
+vsnprintf_(char *restrict s, int size, const char *restrict fmt, va_list ap)
+{
+	int  len;
+
+	len = vsnprintf(s, size, fmt, ap);
+	if (len >= size)
```

Comment:
Test for `len` being negative first. If it is not, `len` should be casted to `size_t` for comparison.